toolchain/sem_ir/stringify_type.cpp

```diff
@@ -283,6 +312,57 @@ auto StringifyTypeExpr(const SemIR::File& sem_ir, InstId outer_inst_id)
         }
         break;
       }
+      case CARBON_KIND(InterfaceWitnessAccess inst): {
+        auto witness_inst_id =
+            sem_ir.constant_values().GetConstantInstId(inst.witness_id);
+        auto witness =
+            sem_ir.insts().GetAs<FacetAccessWitness>(witness_inst_id);
+        auto witness_type_id =
+            sem_ir.insts().Get(witness.facet_value_inst_id).type_id();
+        auto facet_type = sem_ir.types().GetAs<FacetType>(witness_type_id);
+        // TODO: Support != 1 interface better.
+        if (auto impls_constraint = sem_ir.facet_types()
+                                        .Get(facet_type.facet_type_id)
+                                        .TryAsSingleInterface()) {
+          step_stack.PushString(")");
+          auto interface =
+              sem_ir.interfaces().Get(impls_constraint->interface_id);
+          auto entities =
+              sem_ir.inst_blocks().Get(interface.associated_entities_id);
+          size_t index = inst.index.index;
+          CARBON_CHECK(index < entities.size(), "Access out of bounds.");
+          auto entity_inst_id = entities[index];
+          if (auto associated_const =
+                  sem_ir.insts().TryGetAs<AssociatedConstantDecl>(
+                      entity_inst_id)) {
+            step_stack.PushNameId(associated_const->name_id);
+          } else if (auto function_decl = sem_ir.insts().TryGetAs<FunctionDecl>(
+                         entity_inst_id)) {
+            auto function = sem_ir.functions().Get(function_decl->function_id);
+            step_stack.PushNameId(function.name_id);
+          } else {
+            step_stack.PushInstId(entity_inst_id);
+          }
+          step_stack.PushString(".");
+          step_stack.PushNameId(interface.name_id);
+          step_stack.PushString(".(");
+        } else {
+          step_stack.PushTypeId(witness_type_id);
```

Comment:
I think you're missing a `)` here. Should the `PushString` on line 327 be outside the `if`?